app/jenkins-for-jira-ui/src/components/ConnectionPanel/SetUpGuide.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react';
+import { cx } from '@emotion/css';
+import Drawer from '@atlaskit/drawer';
+import PeopleGroup from '@atlaskit/icon/glyph/people-group';
+import {
+	setUpGuideLink,
+	setUpGuideInfoPanel,
+	setUpGuideNestedOrderedList,
+	setUpGuideNestedOrderedListItem,
+	setUpGuideOrderedList,
+	setUpGuideOrderedListItem,
+	setUpGuideOrderListItemHeader,
+	setUpGuideParagraph, setUpGuideContainer
+} from './ConnectionPanel.styles';
+
+type SetUpGuideLinkProps = {
+	onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+	label: string,
+};
+
+const SetUpGuideLink = ({ onClick, label }: SetUpGuideLinkProps): JSX.Element => {
+	return (
+		<button className={cx(setUpGuideLink)} onClick={onClick}>
+			{label}
+		</button>
+	);
+};
+
+type SetUpGuidePipelineStepInstructionProps = {
+	eventType: string,
+	onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+	pipelineStepLabel: string
+};
+
+const SetUpGuidePipelineStepInstruction = ({
+	eventType,
+	onClick,
+	pipelineStepLabel
+}: SetUpGuidePipelineStepInstructionProps): JSX.Element => {
+	return (
+		<p>Add a &nbsp;
+			<SetUpGuideLink onClick={onClick} label={pipelineStepLabel} />&nbsp;
+			step to the end of {eventType} stages.
+		</p>
+	);
+};
+
+enum PipelineEventType {
+	BUILD = 'build',
+	DEPLOYMENT = 'deployment'
+}
+
+type SetUpGuideInstructionsProps = {
+	onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+	eventType: PipelineEventType,
+	globalSettings: boolean,
+	buildFilters: boolean
+};
+
+const SetUpGuideInstructions = ({
+	onClick,
+	eventType,
+	globalSettings,
+	buildFilters
+}: SetUpGuideInstructionsProps): JSX.Element => {
+	const pipelineStepLabel = eventType === PipelineEventType.BUILD ? 'jiraSendBuildInfo' : 'jiraSendDeploymentInfo';
+
```

Comment:
its late and ill reread this in the morning... but it might be nice to break the conditional into a const for '!buildFilters && eventType === PipelineEventType.BUILD' with a lovely variable name ofcourse.
and if you make it a positive test
`globalSettings && !buildFilters && eventType === PipelineEventType.BUILD`

You should be able to get away without duplicating `SetUpGuidePipelineStepInstruction`
and then just terny the OR blah blah bit with !globalSettings

But also, if this works already 🤷